README.md

```diff
@@ -35,33 +35,56 @@ async fn main() {
     let mut runner = Runner::new(
         "https://gitlab.example.com".try_into().unwrap(),
         "runner token".to_owned(),
+        "system id".to_owned(),
         PathBuf::from("/tmp"));
     runner.run(move | _job | async move { Ok(Run{})  }, 16).await.unwrap();
 }
 ```
 
-## Gitlab runner registration
+## Gitlab runner creation
 
-This crate does not support registering new runners with the gitlab server, so this has to be
-done by hand using the gitlab
-[runner registration API](https://docs.gitlab.com/ee/api/runners.html#register-a-new-runner).
+This crate does not support creating new runners on the GitLab server. This can
+be done using the
+[runner creation API](https://docs.gitlab.com/ee/api/users.html#create-a-runner-linked-to-a-user),
+or manually in the GitLab
+[runner management web interface](https://docs.gitlab.com/ee/ci/runners/runners_scope.html).
+Make sure to follow the runner creation with an authentication token workflow,
+as the registration token workflow is deprecated.
+
+One key parameter provided when creating the runner is `run_untagged=false` (or
+leaving the `Run untagged jobs` box unchecked in the web interface), which will
+make the runner *only* pickup jobs which matches its tags. This is important to
+prevent the runner from picking up "normal" jobs which it will not be able to
+process.
+
+When the runner is created GitLab provides an authentication token starting
+with `glrt-`. This token should be provided to the runner for its GitLab
+connection, along with a system ID that identifies the machine on which the
+runner is executed.
+
+The system ID should be a unique string. GitLab doesn't currently require any
+particular formatting, but it is recommended to follow the way the official
+`gitlab-runner` creates system IDs:
```

Comment:
The crate should probably be the one creating this system_id  rather then requiring the user (either end of runner implementation) to create it themselves;  

In our typical setup we run these runners in kubernetes where it should really be unique per container instance. 

I do wonder how/if gitlab garbage collects instances

Reply:
I'm fine creating it in the gitlab-runner-rs crate. That gets beyond my very limited rust capabilities though, would you be able to give it a go ? The logic from the official gitlab-runner can be found in https://gitlab.com/gitlab-org/gitlab-runner/-/blob/main/common/system_id_state.go?ref_type=heads#L89

According to https://docs.gitlab.com/ee/architecture/blueprints/runner_tokens/#ci_runner_machines-record-lifetime, entries in the ci_runner_machines table are automatically cleaned 7 days after the last contact from the respective runner. That page also explains how the system ID is stored in a `.runner_system_id` file, separate from the main runner configuration file.

Reply:
Thanks for the pointers; was just digging through the gitlab code and it's indeed implement as mentioned (once an hour it cleans out all stale machines so 7 days or older). Also as you mentioned gitlab doesn't require a specific format, but it will check the id is less then 64 characters

Reply:
fwiw yeah happy to add the autogeneration; should be pretty simple

Reply:
@pinchartl  did the generation in PR #24 as it now happens behind the scenes most of your changes were dropped; but i did pick up your documentation change as a seperate patch with you as the author :)